src/Common/DeleteCINodeButton/DeleteCINodeButton.tsx

```diff
@@ -0,0 +1,136 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState } from 'react'
+import { ERROR_STATUS_CODE } from '@Common/Constants'
+import { Button, ButtonStyleType, ButtonVariantType, DeleteConfirmationModal } from '@Shared/Components'
+import { ComponentSizeType, DeleteComponentsName } from '@Shared/constants'
+import { ToastManager, ToastVariantType } from '@Shared/index'
+import { ReactComponent as ICDelete } from '@IconsV2/ic-delete.svg'
+import { deleteWorkflow, savePipeline } from './utils'
+import { DeleteCINodeButtonProps } from './types'
+import { preventDefault, showError, stopPropagation } from '..'
+
+export const DeleteCINodeButton = ({
+    testId,
+    isCIPipeline,
+    disabled,
+    title,
+    isJobView,
+    deletePayloadConfig,
+    onDelete,
+    getWorkflows,
+}: DeleteCINodeButtonProps) => {
+    const [showDeleteModal, setShowDeleteModal] = useState<boolean>(false)
+
+    const closeCIDeleteModal = (): void => {
+        setShowDeleteModal(false)
+    }
+
+    const onClickDeleteShowModal = (e) => {
+        stopPropagation(e)
+        preventDefault(e)
+        setShowDeleteModal(true)
+    }
+
+    const onDeleteWorkflow = async () => {
+        await deleteWorkflow(String(deletePayloadConfig.appId), Number(deletePayloadConfig.appWorkflowId))
+            .then((response) => {
+                if (response.errors) {
+                    const { errors } = response
+                    const { userMessage } = errors[0]
+                    ToastManager.showToast({
+                        variant: ToastVariantType.error,
+                        description: userMessage,
+                    })
+                    return
+                }
+
+                if (response.status.toLowerCase() === 'ok') {
+                    ToastManager.showToast({
+                        variant: ToastVariantType.success,
+                        description: 'Workflow Deleted',
+                    })
+                    getWorkflows()
+                }
+            })
+            .catch((errors) => {
+                showError(errors)
+            })
+    }
+
+    const onClickDelete = async () => {
+        const deletePayload = {
+            action: 2, // To delete the pipeline
+            appId: Number(deletePayloadConfig.appId),
+            appWorkflowId: Number(deletePayloadConfig.appWorkflowId),
+            ciPipeline: {
+                id: Number(deletePayloadConfig.pipelineId),
+                name: deletePayloadConfig.pipelineName,
+            },
+        }
+        await savePipeline(deletePayload)
+        if (typeof onDelete === 'function') {
+            onDelete()
```

Comment:
Should this be a promise?

Reply:
No, that will be a function with a regular closing function.